### PR TITLE
feat: Fixing Mantis

### PR DESCRIPTION
**Pull request by giacuong171**
- Testing mantis
- Rewrite pipeline

### CODE REVIEW

File: snappy_pipeline/workflows/somatic_msi_calling/__init__.py [hunk 1]
```diff
@@ -60,30 +61,48 @@
...
 __author__ = "Clemens Messerschmidt <clemens.messerschmidt@bih-charite.de>"
 
+#: Extensions of files to create as main payload
+EXT_VALUES = (".results.txt", ".results.txt.status", ".results.txt.md5", ".results.txt.status.md5")
+
+#: Names of the files to create for the extension
+EXT_NAMES = ("result", "status", "result_md5", "status_md5")
+
+EXT_MATCHED = {
+    "mantis_msi2": {
```
Comment (ericblanc20) [inline snappy_pipeline/workflows/somatic_msi_calling/__init__.py:71]: What is the format of the output? What is `result`, and what is `status`? Ideally, you want a 3rd output containing the minimal MSI/not MSI status in a format independent of the caller. That way, downstream steps can collect the main information without having to rely on the tool's specifics.

**Review by ericblanc20 — CHANGES_REQUESTED** [review]

File: snappy_pipeline/workflows/somatic_msi_calling/__init__.py [hunk 2]
```diff
@@ -119,24 +136,38 @@ def input_function(wildcards):
                 "tumor_bai": ngs_mapping(tumor_base_path + ".bam.bai"),
             }
 
-        assert action == "run", "Unsupported actions"
         return input_function
 
-    @dictify
+    def get_normal_lib_name(self, wildcards):
```
[the remark this answers is not in the document]
Comment (giacuong171) [reply snappy_pipeline/workflows/somatic_msi_calling/__init__.py:141]: mantis does use normal samples, as you can see in the wrapper.

File: snappy_pipeline/workflows/somatic_msi_calling/__init__.py [hunk 3]
```diff
@@ -60,30 +61,48 @@
...
+
 #: Default configuration for the somatic_msi_calling step
 DEFAULT_CONFIG = r"""
 # Default configuration somatic_msi_calling
 step_config:
   somatic_msi_calling:
-    tools: ['mantis']  # REQUIRED - available: 'mantis'
+    tools: ['mantis_msi2']  # REQUIRED - available: 'mantis_msi2'
     path_ngs_mapping: ../ngs_mapping  # REQUIRED
     loci_bed: /fast/groups/cubi/projects/biotools/Mantis/appData/hg19/loci.bed  # REQUIRED
```
[the remark this answers is not in the document]
Comment (giacuong171) [reply snappy_pipeline/workflows/somatic_msi_calling/__init__.py:90]: Yes. Should i provide it in the default config ?

[the remark this answers is not in the document]
Comment (ericblanc20) [reply snappy_pipeline/workflows/somatic_msi_calling/__init__.py:90]: I'd say: 
```
loci_bed: "" # For example: /fast/groups/cubi/projects/biotools/Mantis/appData/hg19/loci.bed for hg19 (=GRCh37 with chr prefix)
             # /fast/groups/cubi/projects/biotools/Mantis/appData.... for GRCh38
```

**Review by ericblanc20 — APPROVED** [review]

**Pull request merged**